Form message from server.

anthonythorne opened the pull request:
The global message returned from the server. Filters allowing the theme to manipulate the html and/or field.
I feel there is a minimum of 'success' and 'error' type messages, giving them a type allows the rendered HTML to reflect this.

Tested in theme altering the message HTML via the filter.

Note: I am not happy directly setting the $_POST param as I am used to adding it to return the response. Open to suggestions.

plugins/wp-chargify/includes/chargify/endpoints/subscriptions.php lines 62 and 65

anthonythorne committed:
Global message returned from the server. Filters allowing the theme to manipulate the html and/or field.

BronsonQuick reviewed (approved):
Nice work Sir! I can understand when you weren't too happy with modifying the approach but the only alternatives I could think of would be using a session or creating a unique transient for the user neither of which seem like a great approach either unless you disagree.

anthonythorne committed:
Update the method to apply the form message, for ease of use within other stages of submission.

anthonythorne commented:
@BronsonQuick I think its mostly because I am used to a OOO approach storing the response for return which can standardize the format of the message array.

Instead, I have added a function to apply and retrieve the messages. I am happy with that for now if you are?

Some checks are still running above.
